qase-playwright/src/playwright.ts

```diff
@@ -1,8 +1,208 @@
+import test from '@playwright/test';
+import { v4 as uuidv4 } from 'uuid';
+import { PlaywrightQaseReporter } from './reporter';
+import * as path from 'path';
+
+export const ReporterContentType = 'application/qase.metadata+json';
+const defaultContentType = 'application/octet-stream';
+
+export interface MetadataMessage {
+  ids?: number[];
+  title?: string;
+  fields?: Map<string, string>;
+}
+
+/**
+ * @class Qase
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class Qase {
+  /**
+   * Add metadata to the test case
+   * @param metadata
+   * @private
+   */
+  private static async addMetadata(metadata: MetadataMessage) {
+    await test.info().attach('qase-metadata.json', {
+      contentType: ReporterContentType,
+      body: Buffer.from(JSON.stringify(metadata), 'utf8'),
+    });
+  }
+
+  /**
+   * Set title for the test case
+   * @param {string} value
+   * @example
+   * test('test', async ({ page }) => {
+   *    Qase.Title("Title");
+   *    await page.goto('https://example.com');
+   * });
+   */
+  static Title(value: string) {
+    this.addMetadata({
+      title: value,
+    }).catch(() => {/**/
+    });
+  }
+
+  /**
+   * Set IDs for the test case
+   *
+   * @param {number | number[]} value
+   *
+   * @example
+   * test('test', async ({ page }) => {
+   *    Qase.Id(1);
+   *    await page.goto('https://example.com');
+   * });
+   *
+   */
+  static Id(value: number | number[]) {
+    this.addMetadata({
+      ids: Array.isArray(value) ? value : [value],
+    }).catch(() => {/**/
+    });
+  }
+
+  /**
+   * Set fields for the test case
+   * @param {Map<string, string>[]} value
+   * @example
+   * test('test', async ({ page }) => {
+   *    Qase.Fields({ 'severity': 'high', 'priority': "medium" });
+   *    await page.goto('https://example.com');
+   * });
+   */
+  static Fields(value: Map<string, string>) {
+    this.addMetadata({
+      fields: value,
+    }).catch(() => {/**/
+    });
+  }
+
+  /**
+   * Attach file to the test case
+   * @param attach
+   * @example
+   * test('test', async ({ page }) => {
+   *   Qase.Attach({ name: 'attachment.txt', content: 'Hello, world!', contentType: 'text/plain' });
+   *   Qase.Attach({ paths: '/path/to/file'});
+   *   Qase.Attach({ paths: ['/path/to/file', '/path/to/another/file']});
+   *   await page.goto('https://example.com');
+   *  });
+   */
+  static Attach(attach: {
```

Comment:
are you sure this method should be named Attach and not attach?

Reply:
yes, because other methods start with a capital letter

Reply:
fixed